inferno-vc/src/Inferno/VersionControl/Operations/Filesystem.hs

```diff
@@ -316,20 +330,35 @@ instance (MonadIO m, MonadMask m, AsType VCStoreError err) => InfernoVCOperation
     let fp = storePath </> "deps" </> show h
     readVCObjectHashTxt fp
 
-  getAllHeads = do
-    VCStorePath storePath <- getTyped <$> ask
-    -- We don't need a lock here because this only lists the heads/ directory, it doesn't
-    -- read any file contents (and I assume the `ls` is atomic)
-    headsRaw <- liftIO $ getDirectoryContents $ storePath </> "heads"
-    pure $
-      foldr
-        ( \hd xs ->
-            case (either (const Nothing) Just $ Base64.decode $ Char8.pack hd) >>= digestFromByteString of
-              Just hsh -> (VCObjectHash hsh) : xs
-              Nothing -> xs
-        )
-        []
-        (map takeFileName headsRaw)
+  deleteAutosavedVCObjectsOlderThan t = do
```

Comment:
sqlite backend can now do this with an efficient `DELETE FROM autosaved_objects WHERE timestamp < ?`